sycl/source/detail/global_handler.hpp

```diff
@@ -0,0 +1,76 @@
+//==--------- global_handler.hpp --- Global objects handler ----------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <CL/sycl/detail/spinlock.hpp>
+#include <CL/sycl/detail/util.hpp>
+#include <detail/platform_impl.hpp>
+#include <detail/plugin.hpp>
+#include <detail/program_manager/program_manager.hpp>
+#include <detail/scheduler/scheduler.hpp>
```

Comment:
Moving these include directives to the source file should improve compilation time.
I think forward declarations would be enough for the header file.